status/reporters.go

```diff
@@ -0,0 +1,203 @@
+
+// WithAppClientCreator is an option which allows the replacement of the github App client constructor function.
+func WithAppClientCreator(creator github.AppClientCreator) GitHubPipelineRunReporterOption {
+	return func(r *GitHubPipelineRunReporter) {
+		r.appClientCreator = creator
+	}
+}
+
+// NewGitHubPipelineRunReporter returns a struct implementing the Reporter interface for GitHub
+func NewGitHubPipelineRunReporter(ctx context.Context, logger logr.Logger, k8sClient client.Reader, pipelineRun *tektonv1beta1.PipelineRun, opts ...GitHubPipelineRunReporterOption) (Reporter, error) {
+	var err error
+	reporter := GitHubPipelineRunReporter{
+		logger:           logger,
+		pipelineRun:      pipelineRun,
+		appClientCreator: github.NewAppClient,
+	}
+
+	for _, opt := range opts {
+		opt(&reporter)
+	}
+
+	pacSecret := v1.Secret{}
+	err = k8sClient.Get(ctx, types.NamespacedName{Namespace: "pipelines-as-code", Name: "pipelines-as-code-secret"}, &pacSecret)
+	if err != nil {
+		return nil, err
+	}
+
+	if helpers.HasAnnotation(pipelineRun, gitops.PipelineAsCodeInstallationIDAnnotation) {
+		reporter.installationID, err = strconv.ParseInt(pipelineRun.GetAnnotations()[gitops.PipelineAsCodeInstallationIDAnnotation], 10, 64)
+		if err != nil {
+			return nil, err
+		}
+
+		ghAppIDBytes, found := pacSecret.Data["github-application-id"]
+		if !found {
+			return nil, errors.New("failed to find github-application-id secret key")
+		}
+
+		reporter.appID, err = strconv.ParseInt(string(ghAppIDBytes), 10, 64)
+		if err != nil {
+			return nil, err
+		}
+
+		reporter.privateKey, found = pacSecret.Data["github-private-key"]
+		if !found {
+			return nil, errors.New("failed to find github-private-key secret key")
+		}
+	}
+
+	return &reporter, nil
+}
+
+func (r *GitHubPipelineRunReporter) createCheckRunAdapter() (*github.CheckRunAdapter, error) {
+	scenario, found := r.pipelineRun.GetLabels()[gitops.ApplicationSnapshotTestScenarioLabel]
+	if !found {
+		return nil, fmt.Errorf("PipelineRun label not found %q", gitops.ApplicationSnapshotTestScenarioLabel)
+	}
+
+	component, found := r.pipelineRun.GetLabels()[gitops.ApplicationSnapshotComponentLabel]
+	if !found {
+		return nil, fmt.Errorf("PipelineRun label not found %q", gitops.ApplicationSnapshotComponentLabel)
+	}
+
+	owner, found := r.pipelineRun.Labels[gitops.PipelineAsCodeURLOrgLabel]
+	if !found {
+		return nil, fmt.Errorf("PipelineRun label not found %q", gitops.PipelineAsCodeURLOrgLabel)
+	}
+
+	repo, found := r.pipelineRun.Labels[gitops.PipelineAsCodeURLRepositoryLabel]
+	if !found {
+		return nil, fmt.Errorf("PipelineRun label not found %q", gitops.PipelineAsCodeURLRepositoryLabel)
+	}
+
+	SHA, found := r.pipelineRun.Labels[gitops.PipelineAsCodeSHALabel]
+	if !found {
+		return nil, fmt.Errorf("PipelineRun label not found %q", gitops.PipelineAsCodeSHALabel)
+	}
+
+	var title, conclusion string
+	succeeded := r.pipelineRun.Status.GetCondition(apis.ConditionSucceeded)
```

Comment:
The Tekton pipelineRun `Succeeded` condition is misleading in our case, because that only means that the pipeline finished with no unexpected errors.
For Integration pipelineRuns (and all other HACBS ones), all pipelines are expected to succeed, and the decision if the tests succeeded is based on the individual TaskRun results. In our case we get those with `GetHACBSTestResultsFromPipelineRun` and then find out if the Integration test passed with `CalculateIntegrationPipelineRunOutcome`.

Reply:
Fixed in fb52df37823ee4f205821cea4dcc85ab7d028e03.

Reply:
Thanks, the changes look good